crates/store/src/key/context.rs

```diff
@@ -62,9 +62,12 @@ impl KeyComponent for PublicKey {
 pub struct ContextIdentity(Key<(ContextId, PublicKey)>);
 
 impl ContextIdentity {
-    pub fn new(context_id: calimero_primitives::context::ContextId, context_pk: [u8; 32]) -> Self {
+    pub fn new(
+        context_id: calimero_primitives::context::ContextId,
+        context_pk: calimero_primitives::identity::PublicKey,
+    ) -> Self {
         Self(Key(
-            GenericArray::from(*context_id).concat(context_pk.into())
+            GenericArray::from(*context_id).concat(GenericArray::from(context_pk.0))
```

Comment:
```suggestion
            GenericArray::from(*context_id).concat(GenericArray::from(*context_pk))
```

Reply:
No worries! I was trying to keep with the existing interface, as every single other struct in that file exposes its fields as public. But I have no problem with the change, in principle. The trouble is, this cannot be dereferenced here in the manner suggested, so is there an alternative I'm missing...?

Reply:
> this cannot be dereferenced here in the manner suggested, so is there an alternative I'm missing

following the clarifications made above, matching the interface for `ContextId` will make this possible